### DISCUSSION TITLE
Resize now work on "freeform" layout

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1365] [Q&A]
Hi @tortmayr
   When I set my root node layout to 'vbox' or 'hbox' it works ok, but when I put it to "freeform" I can not resize my view, Do I miss some setting? I am unsure if this bug or "freeform" is designed to work like this.
```
    protected noPaddingOptions = {
        resizeContainer: true,
        paddingTop: 0,
        paddingBottom: 0,
        paddingLeft: 0,
        paddingRight: 0,
        hGap: 0,
        vGap: 0,
    };

    protected createChipNode(model: Chip): GNode {
        const chipNode = GNode.builder() //
            .id(model.id)
            .type('Chip')
            .layout('freeform')
            .addCssClass('vchip-node')
            .position(model.position)
            .addLayoutOptions(this.noPaddingOptions)
            .addLayoutOptions({ prefWidth: model.size.width, prefHeight: model.size.height });

        const chipContentNode = GNode.builder()
            .type('ChipContent')
            .addCssClass('chip-content')
            .size(model.size.width, model.size.height)
            .addLayoutOptions(this.noPaddingOptions);

        chipNode.add(chipContentNode.build());
        return chipNode.build();
    }

    configureModelElement(context, 'Chip', ChipNode, RoundedCornerNodeView, {
        disable: [connectableFeature],
        enable: [layoutableChildFeature, layoutContainerFeature],
    });
    configureModelElement(context, 'ChipContent', GNode, RectangularNodeView, {
        disable: [connectableFeature, moveFeature, selectFeature],
        enable: [layoutableChildFeature, layoutContainerFeature],
    });
```

![2024-07-10_16-57-55](https://github.com/eclipse-glsp/glsp/assets/4537809/7e54100e-f375-4c13-9a9e-652d99ebf696)

**Comment by tortmayr** [comment]
Hi @wss29,
I'm not aware of any limitations of `freeform` in this regard. So I think your code snippet is expected to work.
I can look into this next week.

**Reply by wss29** [reply]
Hi @tortmayr
I want to implement my model in glsp like sprotty random-graph demo. 
![image](https://github.com/user-attachments/assets/df194d8c-17ae-4c61-99d7-056875cb47c6)
![image](https://github.com/user-attachments/assets/698c0e5a-c87a-4f7d-9725-a5397c8d874a)
1. I also want to add ports to the top and bottom, not just the left and right of the node. and resize the node dynamically in glsp. 
2. After trying many  hours I still did not  work out how to use layout to implement it and resizable, I have seen [1069](https://github.com/eclipse-glsp/glsp/discussions/1069) may have had the same problem
